### PR TITLE
Add esp32 continuous integration.

### PR DESCRIPTION
**Pull request by floitsch**
Currently just builds the esp32 image.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -140,3 +140,55 @@ jobs:
...
+
+      - name: Install dependencies - Linux
+        run: |
+          sudo apt-get update
+          sudo apt-get install ninja-build ccache
+          ninja --version
+          cmake --version
+          gcc --version
+
+      - name: Install Esp32 tools
```
Comment (kasperl) [inline .github/workflows/ci.yml:178]: ESP32?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -140,3 +140,55 @@ jobs:
...
+      # We create a new id every second.
+      # The 'restore-keys' below will make sure that we continue using earlier versions.
+      - name: Create timestamp
+        id: timestamp
+        shell: cmake -P {0}
+        run: |
+          string(TIMESTAMP current_date "%Y-%m-%d-%H;%M;%S" UTC)
+          message("::set-output name=timestamp::${current_date}")
+
+      - name: Ccache Cache
```
Comment (kasperl) [inline .github/workflows/ci.yml:163]: Ccache cache

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (floitsch) [reply .github/workflows/ci.yml]: done.
Also for the original entry.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (floitsch) [reply .github/workflows/ci.yml]: done.

**Pull request merged**